internal/server/handlers/k8s/kubernetes.go

```diff
@@ -0,0 +1,136 @@
+// Package k8s contains the Kubernetes handler.
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/AlekSi/pointer"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+
+	"github.com/percona/everest/api"
+	"github.com/percona/everest/pkg/common"
+	"github.com/percona/everest/pkg/kubernetes"
+	"github.com/percona/everest/pkg/rbac"
+)
+
+const (
+	annotationStorageClassDefault = "storageclass.kubernetes.io/is-default-class"
+)
+
+func (h *k8sHandler) GetKubernetesClusterResources(ctx context.Context) (*api.KubernetesClusterResources, error) {
+	// Get cluster type
+	clusterType, err := h.kubeClient.GetClusterType(ctx)
+	if err != nil {
+		// Instead of failing we switch to a generic cluster type.
+		clusterType = kubernetes.ClusterTypeGeneric
+	}
+
+	var volumes *corev1.PersistentVolumeList
+	if clusterType == kubernetes.ClusterTypeEKS {
+		volumes, err = h.kubeClient.GetPersistentVolumes(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to GetPersistentVolumes: %w", err)
+		}
+	}
+
+	res, err := h.calculateClusterResources(ctx, h.kubeClient, clusterType, volumes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculateClusterResources: %w", err)
+	}
+	return res, nil
+}
+
+func (h *k8sHandler) GetKubernetesClusterInfo(ctx context.Context) (*api.KubernetesClusterInfo, error) {
+	clusterType, err := h.kubeClient.GetClusterType(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to GetClusterType: %w", err)
+	}
+	storagesList, err := h.kubeClient.GetStorageClasses(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to GetStorageClasses: %w", err)
+	}
+	classNames := storageClasses(storagesList)
+	return &api.KubernetesClusterInfo{ClusterType: string(clusterType), StorageClassNames: classNames}, nil
+}
+
+func (h *k8sHandler) GetUserPermissions(ctx context.Context, _ string) (*api.UserPermissions, error) {
+	cm, err := h.kubeClient.GetConfigMap(ctx, common.SystemNamespace, common.EverestRBACConfigMapName)
+	if err != nil {
+		return nil, errors.Join(err, errors.New("could not get Everest RBAC ConfigMap"))
+	}
+	enabled := rbac.IsEnabled(cm)
+	return &api.UserPermissions{
+		Enabled: enabled,
+	}, nil
+}
```

Comment:
I felt a bit confused that the other handlers had these functions in the `handler.go` and the `k8s` handler didn't.

Reply:
There's no good reason tbh. The `k8s` handler has an actual implementation, so I created a separate file. The others don't have any implementation (other than calling `next`) so I didn't create a file.
I can move them to a separate `kubernetes.go` file for consistency ...

Reply:
Updated